CMake/FindMKL.cmake

```diff
@@ -2,42 +2,105 @@
 # This needs a lot of work to make it robust
 INCLUDE( CheckCXXSourceCompiles )
 
+if ( NOT CMAKE_CXX_COMPILER_ID MATCHES "Intel" )
+  # if not intel and MKL_ROOT not set
+  if (NOT MKL_ROOT)
+    find_path(MKL_ROOT "mkl.h"
+      HINTS "${MKLROOT} ${MKL_HOME} $ENV{MKLROOT} $ENV{MKL_ROOT} $ENV{MKL_HOME}")
+    string(REPLACE "/include" "" MKL_ROOT ${MKL_ROOT})
+  endif (NOT MKL_ROOT)
+  if (NOT MKL_ROOT)
+    message (FATAL_ERROR "ENABLE_OUTSIDE_MKL is TRUE and mkl not found. Set MKL_ROOT." )
+  endif (NOT MKL_ROOT)
+
+  #Do we support ilp?
```

Comment:
No, we don't use ILP64.  I don't think there are any really long vectors that need 64 integers for indices.

Reply:
Ok well then I think these libraries should be fairly portable.  They work on Centos and Darwin at least.